Wikipedia/Code/TalkPagesController.swift

```diff
@@ -299,21 +279,34 @@ private extension TalkPageController {
         fetcher.fetchTalkPage(urlTitle: urlTitle, displayTitle: displayTitle, host: host, languageCode: languageCode, revisionID: revisionID, completion: completion)
     }
     
-    func fetchAndUpdate(localTalkPage: TalkPage, revisionID: Int, completion: ((Result<TalkPage, Error>) -> Void)? = nil) {
-        
+    func fetchAndUpdateLocalTalkPage(with moid: NSManagedObjectID, revisionID: Int, completion: ((Result<TalkPage, Error>) -> Void)? = nil) {
         fetchTalkPage(revisionID: revisionID) { (result) in
             DispatchQueue.main.async {
```

Comment:
This DispatchQueue.main.async we could probably remove since perform switches you below anyways.

Reply:
This one's for ensuring the `viewContext` is accessed from the main thread

Reply:
aah I see 👍